### PR TITLE
arch/arm64: remove struct regs_context

### PR DESCRIPTION
**Pull request by lipengfei28**
1.arch/arm64: remove struct regs_context

## Summary

*Update this section with information on why change is necessary,
 what it exactly does and how, if new feature shows up, provide
 references (dependencies, similar problems and solutions), etc.*

## Impact

*Update this section, where applicable, on how change affects users,
 build process, hardware, documentation, security, compatibility, etc.*

## Testing

*Update this section with details on how did you verify the change,
 what Host was used for build (OS, CPU, compiler, ..), what Target was
 used for verification (arch, board:config, ..), etc. Providing build
 and runtime logs from before and after change is highly appreciated.*

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



## NuttX PR Requirements Review

The provided PR description **does not** meet the NuttX requirements.  Here's why and how to improve it:

**Missing Information:**

* **Summary:** 
    * **Why is this change necessary?**  Is `struct regs_context` obsolete?  Does it cause conflicts?  Provide context!
    * **What exactly does removing it achieve?**  Improved performance? Code simplification? 
    * **How was the removal implemented?**  Were there dependencies on `regs_context` that needed refactoring? 
* **Impact:**
    * **Potential Impacts are not addressed.**  Consider:
        * **Build:** Will removing `regs_context` cause build errors in specific configurations?
        * **Hardware:** Are there architectures/boards relying on this struct?
        * **Compatibility:** Could this break existing code that used `regs_context`? 
        * **Documentation:**  Does any documentation need updating to reflect this removal? 
* **Testing:**
    * **No testing details provided.** You **must** include:
        * **Build Host:** OS, CPU architecture, compiler version (e.g., Linux, x86_64, GCC 12.1)
        * **Target(s):**  Specific architecture, board, and configuration (e.g., ARMv7-M, STM32F4 Discovery, hello_world config)
    * **Logs are essential:**
        * Show relevant build logs (or excerpts) before and after the change to demonstrate the impact. 
        * If possible, include runtime logs from your target to prove the change works as intended.

**Example of an Improved Description:**

```markdown
# arch/arm64: Remove struct regs_context

## Summary

This PR removes the `struct regs_context` definition from the ARM64 architecture code. 

**Reason for Change:**  The `regs_context` structure was found to be unused throughout the ARM64 codebase. Its presence added unnecessary complexity and increased the binary size marginally. 

**Implementation:** The `regs_context` definition has been removed. References to this structure have been eliminated, ensuring a clean and functional codebase.

## Impact

* **Build:** This change is expected to slightly reduce the final binary size for ARM64 builds. No build errors are anticipated.
* **Hardware:** No impact on hardware.
* **Compatibility:**  This change is potentially breaking.  If any out-of-tree code relied on `regs_context`, it will need to be updated. 
* **Documentation:** The `arch/arm64` documentation will be updated to reflect the removal of `regs_context`. 

## Testing

**Build Host:**
* OS: Ubuntu 22.04
* CPU: x86_64
* Compiler: GCC 11.3.0

**Target:**
* Architecture: ARM64 (QEMU virt machine)
* Board:     qemu-aarch64
* Configuration:  nsh

**Testing Logs Before Change:**
```
(Build log excerpt showing the size of the binary before the change)
```

**Testing Logs After Change:**
```
(Build log excerpt showing the size reduction in the binary)

(Output from NSH on the QEMU target, demonstrating that the system still functions correctly)
``` 
```

**Key Takeaways:**

* Be thorough and descriptive in your PR summary.
* Carefully consider and document all potential impacts of your change.
* Provide detailed testing information and supporting logs.

**Comment by xiaoxiang781216** [comment]
please rebase to fix the conflict

**Comment by lipengfei28** [comment]
> please rebase to fix the conflict

done

File: arch/arm64/src/common/arm64_schedulesigaction.c [hunk 1]
```diff
@@ -39,43 +39,43 @@
...
 
 #if CONFIG_ARCH_ARM64_EXCEPTION_LEVEL == 3
-  psigctx->spsr      = SPSR_MODE_EL3H | DAIF_FIQ_BIT | DAIF_IRQ_BIT;
+  tcb->xcp.regs[REG_SPSR]   = SPSR_MODE_EL3H | DAIF_FIQ_BIT | DAIF_IRQ_BIT;
 #else
-  psigctx->spsr      = SPSR_MODE_EL1H | DAIF_FIQ_BIT | DAIF_IRQ_BIT;
+  tcb->xcp.regs[REG_SPSR]   = SPSR_MODE_EL1H | DAIF_FIQ_BIT | DAIF_IRQ_BIT;
 #endif
-  psigctx->sp_elx    = (uint64_t)stack_ptr;
+
```
Comment (pussuw) [inline arch/arm64/src/common/arm64_schedulesigaction.c:66]: Is sp_elx not supposed to be set here anymore ?

File: arch/arm64/src/common/CMakeLists.txt [hunk 2]
```diff
@@ -63,6 +63,10 @@ endif()
 
 if(CONFIG_ARCH_HAVE_MPU)
   list(APPEND SRCS arm64_mpu.c)
+  set_source_files_properties(
```
Comment (XuNeo) [inline arch/arm64/src/common/CMakeLists.txt:66]: Remove, it's confirmed that no need to exclude arm64_mpu.c

File: arch/arm64/src/common/arm64_fork.c [hunk 3]
```diff
@@ -181,58 +179,56 @@ pid_t arm64_fork(const struct fork_s *context)
...
+  child->cmn.xcp.regs[REG_X21]    = context->regs[FORK_REG_X21];
+  child->cmn.xcp.regs[REG_X22]    = context->regs[FORK_REG_X22];
+  child->cmn.xcp.regs[REG_X23]    = context->regs[FORK_REG_X23];
+  child->cmn.xcp.regs[REG_X24]    = context->regs[FORK_REG_X24];
+  child->cmn.xcp.regs[REG_X25]    = context->regs[FORK_REG_X25];
+  child->cmn.xcp.regs[REG_X26]    = context->regs[FORK_REG_X26];
+  child->cmn.xcp.regs[REG_X27]    = context->regs[FORK_REG_X27];
+  child->cmn.xcp.regs[REG_X28]    = context->regs[FORK_REG_X28];
+  child->cmn.xcp.regs[REG_FP]     = newfp;
+  child->cmn.xcp.regs[REG_SPSR]   = SPSR_MODE_EL1H;
```
Comment (XuNeo) [inline arch/arm64/src/common/arm64_fork.c:212]: Remove. It's set below.

Comment (lipengfei28) [reply arch/arm64/src/common/arm64_fork.c]: done

**Review by pussuw — APPROVED** [review]

**Pull request merged**